app/models/conference.rb

```diff
@@ -87,10 +87,9 @@ def user_registered? user
   def delete_event_schedules
     if self.start_hour_changed? || self.end_hour_changed?
       event_schedules = program.event_schedules.select do |event_schedule|
-        event_schedule.event.program == program &&
-        (event_schedule.start_time.strftime('%H').to_i < start_hour ||
-        event_schedule.end_time.strftime('%H').to_i > end_hour ||
-        (event_schedule.end_time.strftime('%H').to_i == end_hour && event_schedule.end_time.strftime('%M').to_i > 0))
+        event_schedule.start_time.hour < start_hour ||
+        event_schedule.end_time.hour > end_hour ||
+        (event_schedule.end_time.hour == end_hour && event_schedule.end_time.hour > 0)
```

Comment:
I think here you meant `minute` instead of `hour` in the second part: 

`(event_schedule.end_time.hour == end_hour && event_schedule.end_time.minute > 0)`